app/controllers/api/users_controller.rb

```diff
@@ -34,6 +33,17 @@ def create
       end
     end
 
+    def destroy
+      ## ユーザが自分自身だった場合
+      if token_user == user
+        errors = [{ name: 'user_id', message: t('.destroy_self') }]
+        return render 'api/errors', locals: { errors: }, status: :unprocessable_entity
+      end
```

Comment:
この部分、エラー出力する部分をメソッドに切り出すとアクションがシンプルになると思いました。

```suggestion
      render_destory_self_error and return if token_user == user
```

Reply:
#44 のPRとコンフリクトになるかもしれないので別PRで出します